design/open/lakefs-kv-execution-plan.md

```diff
@@ -5,31 +5,32 @@ High level overview of the interface described in https://github.com/treeverse/l
 
 ## Transform from using db to kv package
 
-High level interface described in the kv proposal, we should consider taking the DynamoDB interface to address the way we communicate `set-if` functionality while marshal the same value without fetching data first.
-
-Key/value format can be done using ProtoBuf / JSON.
-
-- key format based on path - a format with escaping should be supported to enable easy way to encode values as part of the key
-- each model should include version as part of the model information
-- migrate from old model to new one without downtime
+High level API described in the kv proposal, we should consider taking the DynamoDB API to address the way we communicate `set-if` functionality while marshal the same value without fetching data first.
+Key/value format can be done using ProtoBuf / JSON - discussion and information will be part of a design document.
+Each data currently stored using 'db' in a table will be migrated to a key value format. This is the first migration that will be supported on postgres. When all data is migrated from the current tables to kv, the migration will support data format changes inside the key/value.
```

Comment:
Bug like any other issue we have today will have to be address before we can continue.
We can change the execution plan to include testing the new 'kv' functionality on all aspects first before migrating each package. This will include to verify that the kv functionality can handle each package requirements.
@itaiad200 WDYT?

Reply:
What do you mean by `all aspects`? 
I'm sure we'll figure out things as we go and we'll have to adapt like fixing already migrated packages implementations, changing the kv interface to support new ones, etc. We need to test with every packages we add but there's no guarantee that nothing won't change beyond that. Currently it states `When all data is migrated from the current tables to kv, the migration will support data format changes inside the key/value.` but I believe we need to have a way to change things for migrated packages even if the migration of all packages isn't complete.

Reply:
Just that :) the ability to fix migrated packages at all times.

Reply:
Line 10 state that "migration will support data format changes...". The point was to make sure that migration of kv data as part of each lakefs version should be supported at the first version of lakeFS with kv. But not during the development while we support migration from db to kv. The kv changes at this step is work in progress / beta / experimentation.